megatron/core/inference/engines/dynamic_engine.py

```diff
@@ -716,6 +723,11 @@ def _add_request(
             request.status = Status.FAILED
             request.add_event_error_nontransient(TokenOverflowError(request_id))
 
+        # Tokenize stop words if provided
+        if request.sampling_params.stop_words:
+            stop_word_ids = [self.controller.tokenize_prompt(stop_word, add_BOS=False) for stop_word in request.sampling_params.stop_words]
+            request.sampling_params.stop_word_ids = stop_word_ids
```

Comment:
I would prefer if `stop_word_ids` was kept directly inside the `request` object, instead of added to the `sampling_params` object.